### PR TITLE
Rook move logic

### PR DESCRIPTION
**Pull request by AUcod3r**
Incorporated Mike's direction.

### CODE REVIEW

File: app/models/piece.rb [hunk 1]
```diff
@@ -2,13 +2,38 @@ class Piece < ActiveRecord::Base
...
   def possible_moves
     []
   end
 
+  def valid_move?(destination)
+    dest_x = destination[:x].to_i
+    dest_y = destination[:y].to_i
+    finaldestination = [dest_x, dest_y]
+    p valid_moves
+    p finaldestination
```
Comment (hoitomt) [inline app/models/piece.rb]: You have a couple of print statements in here. It's not bad for now, but typically we wouldn't want to commit these to the master branch. But no worries for now

**Comment by hoitomt** [comment]
@AUcod3r This looks good to me. Nice work!

**Comment by hoitomt** [comment]
@AUcod3r Are you able to try this with the application? i.e. can you drag a rook to an invalid spot and have it return an error. And can you drag it to a valid spot and have it return success?

**Comment by AUcod3r** [comment]
@hoitomt yes sir. Fully tested it within the app

**Comment by AUcod3r** [comment]
@hoitomt oh, and it won't write into the DB either if the move isn't valid

**Comment by AUcod3r** [comment]
@hoitomt OK, one more point is that the board stays current too :)

**Comment by AUcod3r** [comment]
@hoitomt removed the print statements

**Comment by hoitomt** [comment]
Sweet! Let's get this merged
:+1:

**Pull request merged**